python/ray/util/sgd/data/imagenet_dataset.py

```diff
@@ -0,0 +1,95 @@
+from ray.util.sgd.data.local_dataset import LocalDataset
+from PIL import Image
+import io
+import os
+import sys
+
+
+def pil_loader(raw_bytes):
+    with io.BytesIO(raw_bytes) as f:
+        img = Image.open(f)
+        return img.convert("RGB")
+
+
+# From torch.utils.data.ImageNet
+def _find_classes(dir):
+    """Finds the class folders in a dataset.
+
+    Args:
+        dir (string): Root directory path.
+
+    Returns:
+        tuple: (classes, class_to_idx) where classes are relative to
+    (dir), and class_to_idx is a dictionary.
+
+    Ensures:
+        No class is a subdirectory of another.
+
+    """
+    if sys.version_info >= (3, 5):
+        # Faster and available in Python 3.5 and above
+        classes = [d.name for d in os.scandir(dir) if d.is_dir()]
```

Comment:
Nice! Do we support Python <3.5? I don't see any wheels for it.

Reply:
good catch